internal/tracing/tracing.go

```diff
@@ -71,9 +71,14 @@ func Init(workerID string) (func(ctx context.Context) error, error) {
 
 // TracedHandler attaches a tracing handler to http routes.
 func TracedRoutes(component string, routes map[string]jape.Handler) map[string]jape.Handler {
+	cache := make(map[string]http.Handler)
 	adapt := func(route string, h jape.Handler) jape.Handler {
 		return jape.Adapt(func(h http.Handler) http.Handler {
```

Comment:
Would this be better to handle on the Jape side? It seems like Jape should be able to wrap an `http.Handler` without creating a new handler every time.

Reply:
I found out this is all wrong. It would be better to handle on Jape side indeed **but** I think we are perhaps not using `otelhttp` the way it's meant to be used but I'm not sure yet and had to run. I think we might be able to pass the entire server mux to `NewHandler` and pass a custom naming function. I still have to verify whether I can get the traces to be the same before/after fix. I have to verify it but perhaps doing it that way avoids recreating the handler every time and also avoids having to update Jape. Fixes welcome though if you see an easy way out 🤷‍♂️ just had to run and pushed a DRAFT and then quickly after found out it was all wrong.

Reply:
ok, think I fixed it: https://github.com/SiaFoundation/jape/commit/8a6d592a9bfb9f6861e4ede881ef0ef0a3662b54

That said, if we can call `otelhttp.NewHandler` on the whole `jape.Mux`, I think that would be preferable.

Reply:
Unfortunately we can not easily wrap the entire `jape.Mux` in an `otelhttp.NewHandler` because we want pretty span names (containing the route template and not things like: `bus: POST /contract/fcid:0121bde9cacb733fdbc3dd5feeafd17d22968556ea50ba26f7c073b37f345765/acquire`). 

I think `gorilla/mux` supports something like that (`mux.CurrentRoute(req).GetPathTemplate()`), but `julienschmidt/httprouter` does not and implementing it ourselves seems a bit over the top only to avoid wrapping every route once. Might come with performance tradeoffs as well.

Reply:
Verified that the jape update fixes the memory leak.